### PR TITLE
Adding cert-manager guidance to TSG

### PR DESCRIPTION
**Pull request by heoelri**
This PR contains some additional guidance on how to fix cert-manager.

### CODE REVIEW

File: docs/reference-implementation/Troubleshooting.md [hunk 1]
```diff
@@ -81,6 +81,35 @@ Deployment of service [HealthService | BackgroundProcessor | CatalogService] fai
...
+**Description:** The "Install workload CatalogService on AKS clusters" step actively monitors the provisioning of a LetsEncrypt certificate for the ingress. If the certificate is not ready, or cannot successfully provisioned this task will fail.
+
+**Solution:** This is most of the time caused by hitting LetsEncrypt thresholds (see [Rate Limits](https://letsencrypt.org/docs/rate-limits/) for more details) for certificate provisioning. You can try to re-run the step, or wait for the certificate to be ready.
+
+```console
+# Check certificate status
+kubectl get certificates -n workload
+
+# Check certificate request status
+kubectl get certificaterequests -n workload
```
Comment (sebader) [inline docs/reference-implementation/Troubleshooting.md:105]: how about adding a "k delete certificaterequest" as a possible solution? this should immediately trigger reissue

Comment (heoelri) [reply docs/reference-implementation/Troubleshooting.md:105]: I was thinking about that as well - what's important though is to make sure that the reason is clear - in case you're hitting a rate limit on the Let's Encrypt side you can delete it as often as you want and nothing will change.

And I'd suggest to delete the certificate not the certificaterequest. Let me rephrase it a bit.

File: docs/reference-implementation/Troubleshooting.md [hunk 2]
```diff
@@ -81,6 +81,45 @@ Deployment of service [HealthService | BackgroundProcessor | CatalogService] fai
...
+
+Relevant log events are for example `cert-manager/certificates-trigger "msg"="Not re-issuing certificate as an attempt has been made in the last hour" "key"="workload/catalogservice-ingress-secret" "retry_delay"=3599055978096`.
+
+To manually force cert-manager to re-request the certificate you can delete the existing one:
+
+```console
+# Check certificate status
+kubectl get certificates -n workload
+
+# Delete certificate (in FALSE state)
```
Comment (sebader) [inline docs/reference-implementation/Troubleshooting.md:119]:
```suggestion
# Delete certificate (if in `ready=FALSE` state). This will trigger cert-manager to create a new certificate request
```

File: docs/reference-implementation/Troubleshooting.md [hunk 3]
```diff
@@ -81,6 +81,45 @@ Deployment of service [HealthService | BackgroundProcessor | CatalogService] fai
...
+```
+
+Relevant log events are for example `cert-manager/certificates-trigger "msg"="Not re-issuing certificate as an attempt has been made in the last hour" "key"="workload/catalogservice-ingress-secret" "retry_delay"=3599055978096`.
+
+To manually force cert-manager to re-request the certificate you can delete the existing one:
+
+```console
+# Check certificate status
+kubectl get certificates -n workload
+
```
Comment (sebader) [inline docs/reference-implementation/Troubleshooting.md:118]: I guess we dont have a sample console output right now, that we could show?! maybe lets keep that in mind and add once we run into it again

**Review by sebader — CHANGES_REQUESTED** [review]

**Comment by heoelri** [comment]
Before this comes up - I start questioning if it makes sense to maintain the same/the overlapping TSG content (which is currently most of it) in all RIs.

**Comment by sebader** [comment]
> Before this comes up - I start questioning if it makes sense to maintain the same/the overlapping TSG content (which is currently most of it) in all RIs.

yeah valid point. Since -Connected mostly adds stuff, we could keep the common issues in -Online and have in -Connected a pointer to this and then only maintain issues that are specific to Connected

File: docs/reference-implementation/Troubleshooting.md (hunk 3 above)
Comment (heoelri) [reply docs/reference-implementation/Troubleshooting.md:118]: Right - I've already manually fixed and destroyed the affected environment.

**Review by sebader — APPROVED** [review]

**Comment by sebader** [comment]
thanks for this!
